src/inmanta/ast/statements/assign.py

```diff
@@ -272,17 +272,12 @@ class Assign(AssignStatement):
     provides:      variable
     """
 
-    def __init__(self, name: str, value: ExpressionStatement) -> None:
+    def __init__(self, variable: LocatableString, value: ExpressionStatement) -> None:
         AssignStatement.__init__(self, None, value)
-        self.name = name
+        self.name = variable.name
```

Comment:
If you change this you could just as well have `self.name` be a `LocatableString` as well. Just make sure to check where it's currently used and use `str()` where appropriate.